experimental/OrthogonalDiscriminants/src/utils.jl

```diff
@@ -0,0 +1,103 @@
+@doc raw"""
+    order_omega_mod_N(d::IntegerUnion, q::IntegerUnion, N::IntegerUnion) -> Pair{Bool, Bool}
+
+Return `(flag_plus, flag_minus)` where `flag_plus` and `flag_minus`
+are `true` or `false`, depending on whether `N` divides the order
+of the orthogonal groups $\Omega^+(d, q)$ and $\Omega^-(d, q)$.
+
+# Examples
+```jldoctest
+julia> Oscar.OrthogonalDiscriminants.order_omega_mod_N(4, 2, 60)
+(false, true)
+
+julia> Oscar.OrthogonalDiscriminants.order_omega_mod_N(4, 5, 60)
+(true, true)
+```
+"""
+function order_omega_mod_N(d::IntegerUnion, q::IntegerUnion, N::IntegerUnion)
+  @req is_even(d) "d must be even"
+  m = div(d, 2)
+  exp = 0
+  while mod(N, q) == 0
+    exp = exp + 1
+    N = div(N, q)
+  end
+  facts = collect(factor(q))
+  p = facts[1][1]
+  if mod(N, p) == 0
+    exp = exp + 1
```

Comment:
Huh, so first `exp` was defined as "`q^exp` is largest power of `q` dividing the original `N`. But after this loop, `exp = exp_q + exp_p`, where `original_N = q^exp_q * p^exp_p * N'` and `N'` is coprime to `p`. Is that intentional? It jus seems so unusual.

Reply:
Oh no, I totally misread that: if there are `p` divisors left, you only increment `exp ` once. Nevermind.